### PR TITLE
add `experiment opa` subcommand

### PR DESCRIPTION
**Pull request by jzelinskie**
This change adds an OPA subcommand that has authzed builtin functions added to the OPA runtime.

Example usage:

```
~/C/o/zed ❯❯❯ ./zed experiment opa eval 'authzed.check("user:fred", "reader", "document:firstdoc", "")'
{
  "result": [
    {
      "expressions": [
        {
          "value": true,
          "text": "authzed.check(\"user:fred\", \"reader\", \"document:firstdoc\", \"\")",
          "location": {
            "row": 1,
            "col": 1
          }
        }
      ]
    }
  ]
}
```

### CODE REVIEW

File: cmd/zed/experiment.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+func RegisterAuthzedBuiltins(system string, client *authzed.Client) {
+	rego.RegisterBuiltin4(
+		&rego.Function{
+			Name:    "authzed.check",
+			Decl:    types.NewFunction(types.Args(types.S, types.S, types.S, types.S), types.B),
+			Memoize: false,
+		},
+		func(bctx rego.BuiltinContext, a, b, c, d *ast.Term) (*ast.Term, error) {
```
Comment (josephschorr) [inline cmd/zed/experiment.go:57]: maybe change `a, b, c, d`?

File: cmd/zed/experiment.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+			}
+			if err := ast.As(c.Value, &objectStr); err != nil {
+				return nil, err
+			}
+			if err := ast.As(d.Value, &zedToken); err != nil {
+				return nil, err
+			}
+
+			var subjectNS, subjectID string
+			err := stringz.SplitExact(subjectStr, ":", &subjectNS, &subjectID)
```
Comment (josephschorr) [inline cmd/zed/experiment.go:74]: This also needs to support the implicit `...`

File: cmd/zed/experiment.go [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+			request := &v0.CheckRequest{
+				TestUserset: &v0.ObjectAndRelation{
+					Namespace: stringz.Join("/", system, objectNS),
+					ObjectId:  objectID,
+					Relation:  relation,
+				},
+				User: &v0.User{UserOneof: &v0.User_Userset{Userset: &v0.ObjectAndRelation{
+					Namespace: stringz.Join("/", system, subjectNS),
+					ObjectId:  subjectID,
+					Relation:  "...",
```
Comment (josephschorr) [inline cmd/zed/experiment.go:94]: Make sure this is changed like the other locations

File: internal/storage/config.go [hunk 4]
```diff
@@ -41,7 +41,7 @@ func CurrentToken(cs ConfigStore, ts TokenStore) (Token, error) {
 		return Token{}, err
 	}
 
-	return ts.Get(cfg.CurrentToken, false)
+	return ts.Get(cfg.CurrentToken, true)
```
Comment (josephschorr) [inline internal/storage/config.go:44]: Why this change?

File: cmd/zed/experiment.go (hunk 1 above)
Comment (jzelinskie) [reply cmd/zed/experiment.go]: This was the style used in the example, but I agree it's not very useful.

File: internal/storage/config.go (hunk 4 above)
Comment (jzelinskie) [reply internal/storage/config.go]: It's not used anywhere anymore. The only time we use --reveal-tokens is on list.

**Review by josephschorr — APPROVED** [review]
LGTM
